Bottom: 2f951c94ecca87b2567d9167b3d2559d4ab84935
Top:    f6120f940995e4df89b62dd2ef64f0fdad861ce3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-13 01:56:55 +0000

DRAW_FUNCS: Attempt at different render order for PCB+GL


---

diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index 33bbd9b..a1260a2 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -392,12 +392,25 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   int group = GetLayerGroupNumberByPointer (layer);
   struct poly_info info = {drawn_area, layer};
 
+  if (layer_num < max_copper_layer)
+    {
+      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
+      r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
+    }
+
   /* print the non-clearing polys */
   r_search (layer->polygon_tree, drawn_area, NULL, poly_callback, &info);
 
   if (TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
 
+  if (layer_num < max_copper_layer)
+    {
+      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
+      r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
+    }
+
+  /* draw all visible lines this layer */
   r_search (layer->line_tree, drawn_area, NULL, line_callback, layer);
   r_search (layer->arc_tree,  drawn_area, NULL, arc_callback,  layer);
   r_search (layer->text_tree, drawn_area, NULL, text_callback, layer);
@@ -433,8 +446,6 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
 
   /* draw vias */
   r_search (PCB->Data->via_tree, drawn_area, NULL, via_inlayer_callback, layer);
-  r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
-  r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
 }
 
 struct draw_funcs d_f = {
